node/sc/vt_recovery.go

```diff
@@ -252,19 +256,42 @@ func retrievePendingEventsFrom(hint *valueTransferHint, br *bridge.Bridge) ([]*R
 	}
 
 	var pendingEvents []*RequestValueTransferEvent
-	it, err := br.FilterRequestValueTransfer(&bind.FilterOpts{Start: hint.blockNumber}) // to the current
+
+	curBlkNum, err := bi.GetCurrentBlockNumber()
 	if err != nil {
 		return nil, err
 	}
-	for it.Next() {
-		logger.Trace("pending nonce in the event", "requestNonce", it.Event.RequestNonce)
-		if it.Event.RequestNonce >= hint.handleNonce {
-			logger.Trace("filtered pending nonce", "requestNonce", it.Event.RequestNonce, "handledNonce", hint.handleNonce)
-			pendingEvents = append(pendingEvents, &RequestValueTransferEvent{it.Event})
+
+	done := false
+	startBlkNum := hint.blockNumber
+	endBlkNum := startBlkNum + filterLogsStride
+
+pendingTxLoop:
+	for !done && startBlkNum <= curBlkNum {
+		if endBlkNum > curBlkNum {
+			endBlkNum = curBlkNum
+			done = true
```

Comment:
I think `done` is not needed. 
If `endBlkNum` > `curBlkNum`, `startBlkNum` became `curBlkNum +1` at the end of the for loop.
Therefore the condition of this for loop `startBlkNum <= curBlkNum` contains all cases of `!done`.

Reply:
Fixed.